apis/v1alpha1/httproute_types.go

```diff
@@ -318,13 +333,14 @@ type HTTPRouteMatch struct {
 	// Path specifies a HTTP request path matcher. If this field is not
 	// specified, a default prefix match on the "/" path is provided.
 	//
+	// +optional
 	// +kubebuilder:default={type: "Prefix", value: "/"}
 	Path HTTPPathMatch `json:"path,omitempty"`
 
 	// Headers specifies a HTTP request header matcher.
 	//
 	// +optional
-	Headers *HTTPHeaderMatch `json:"headers"`
+	Headers HTTPHeaderMatch `json:"headers,omitempty"`
```

Comment:
optional should be a pointer

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required

Reply:
> They are a pointer type in the Go definition (e.g. AwesomeFlag *SomeFlag) __or have a built-in nil value (e.g. maps and slices).__

Reply:
I'll revert this change so the PR can proceed.